library/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java

```diff
@@ -1453,6 +1502,10 @@ public void initializeOutput(ExtractorOutput output, int trackId) throws ParserE
         case CODEC_ID_SUBRIP:
           mimeType = MimeTypes.APPLICATION_SUBRIP;
           break;
+        case CODEC_ID_ASS:
+          mimeType = MimeTypes.TEXT_SSA;
+          ssaHeader = codecPrivate;
```

Comment:
Is this the header that you don't want to prefix onto each sample? If so, put it into the format object (by doing: `initializationData = codecPrivate`) and don't prefix it onto any sample at all.